### ISSUE TITLE
Tapping widgets

### ISSUE THREAD

**Issue by yoh-there** [fesch/CanZE#301]
A German user reported that tapping widgets leads to unpredictable results. Sometimes the single, big widget appears, sometimes nothing happens, sometimes the app crashes.

I can confirm the "nothing happens" (S3). I never used this feature myself so that's why this might pop up so late.

http://www.goingelectric.de/forum/renault-zoe-zubehoer/hacking-into-zoes-canbus-t12199-140.html#p296910 and next page

**Comment by fesch** [comment]
When you touch a widget, it open at full-screen.

When you tap it but drag the finger into some direction (slide), nothing happens, because I reserved this sliding movement for some later feature ... not yet implemented.

If the app crashed, there mus be some other problem ...

**Comment by yoh-there** [comment]
I can assure you either this is not (always) true about tapping, or my phones tap-versus slide behaviour is way way way off (it isn't ;-) )

**Comment by yoh-there** [comment]
Maybe this helps. When I tap a widget in "Consumption":
01-07 23:43:10.511 5080-5080/lu.fisch.canze D/  CanZE: WidgetView: maskedAction = 0
01-07 23:43:10.571 5080-5080/lu.fisch.canze D/  CanZE: WidgetView: maskedAction = 2
01-07 23:43:10.571 5080-5080/lu.fisch.canze D/  CanZE: WidgetView: maskedAction = 1

**Comment by yoh-there** [comment]
Yes. I looked at your code. There are many many times "2" (move) events even when I tap very, very, very light

**Comment by yoh-there** [comment]
~~In WidgetView.java I commented out line 263, problem solved. Well, for now. It seems the slide trigger is way too sensitive. I think the move should only be triggered if it's position has at least once been past a trigger distance from the down event.~~

Edit: ignore above

```
            if (Math.abs(downX - event.getX()) + Math.abs(downY - event.getY()) > 20) {
                motionMove=true;
            }
```

**Comment by yoh-there** [comment]
Fixed in pull #302
